### PR TITLE
Fixed some bugs and added the item spawn feature

### PR DESCRIPTION
**Pull request by FoxWorn3365**
yeah @SpGerg is a furry uwuwuw

### CODE REVIEW

**Review by SpGerg — COMMENTED** [review]
all this is unfixable shit. Review is end. hehehehe

File: UncomplicatedCustomItems/API/Features/SummonedCustomItem.cs [hunk 1]
```diff
@@ -161,16 +174,23 @@ internal void OnDrop(DroppedItemEventArgs dropped)
...
-            if (CustomItem.CustomItemType == CustomItemType.Item && itemData.Event == itemEvent)
+            if (CustomItem.CustomItemType == CustomItemType.Item && CustomItem.CustomData is ICustomItem && ((IItemData)CustomItem.CustomData).Event == itemEvent)
             {
+                IItemData Data = CustomItem.CustomData as IItemData;
                 Log.Debug($"Firing events for item {CustomItem.Name}");
-                if (itemData.Command is not null && itemData.Command.Length > 2)
+                if (Data.Command is not null && Data.Command.Length > 2)
                 {
-                    Server.ExecuteCommand(itemData.Command.Replace("%id%", player.Id.ToString()), player.Sender);
+                    if (!Data.Command.Contains("P:"))
```
Comment (SpGerg) [inline UncomplicatedCustomItems/API/Features/SummonedCustomItem.cs:183]: P?

File: UncomplicatedCustomItems/Commands/User/Use.cs [hunk 2]
```diff
@@ -28,6 +30,33 @@ public override bool Execute(ArraySegment<string> arguments, Player player, out
...
+                {
+                    // Error: too few arguments!
+                    response = $"Sorry but this command requires {count} arguments, {arguments.Count} found.";
+                    return false;
+                }
+                else
+                {
+                    for (int i = 0; i < count; i++)
+                    {
+                        int IndexToReplace = Data.Command.IndexOf('#');
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Commands/User/Use.cs:50]: indexToReplace

File: UncomplicatedCustomItems/Commands/User/Use.cs [hunk 3]
```diff
@@ -28,6 +30,33 @@ public override bool Execute(ArraySegment<string> arguments, Player player, out
                 return false;
             }
 
+            // Ok now we have to check if the custom item command contains any & (= args)
+            IItemData Data = Item.CustomItem.CustomData as IItemData;
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Commands/User/Use.cs:34]: data

**Comment by FoxWorn3365** [comment]
perfect, let's merge!

File: UncomplicatedCustomItems/Elements/CustomItem.cs [hunk 4]
```diff
@@ -47,6 +47,12 @@ public class CustomItem : ICustomItem
         [Description("The scale of the custom item, 0 0 0 means disabled")]
         public Vector3 Scale { get; set; } = Vector3.one;
 
+        /// <summary>
+        /// The <see cref="Elements.Spawn"/> settings for the item
+        /// </summary>
+        [Description("The spawn settings for the item")]
+        public ISpawn Spawn { get; set; } = new Spawn();
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Elements/CustomItem.cs:54]: Are you sure this works?

**Review by SpGerg — CHANGES_REQUESTED** [review]

**Pull request merged**